src/test/java/javax/jmdns/impl/ServiceInfoImplTest.java

```diff
@@ -0,0 +1,61 @@
+package javax.jmdns.impl;
+
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.DatagramPacket;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author stefan eicher https://github.com/stefaneicher
+ */
+public class ServiceInfoImplTest {
+
+    @Test
+    public void test_ip_address_is_set() throws Exception {
+        byte[] buf = readFile("a_record_before_srv.bin");
+        DNSIncoming msg = new DNSIncoming(new DatagramPacket(buf, buf.length));
+        JmDNSImpl jmDNS = new JmDNSImpl(null, null);
+        ServiceInfoImpl serviceInfo = new ServiceInfoImpl(
+                "_ibisip_http._tcp.local.",
+                "DeviceManagementService",
+                "",
+                -1,
+                -1,
+                -1,
+                true,
+                new byte[]{});
+        jmDNS.addListener(serviceInfo, null);
+        jmDNS.handleResponse(msg);
+
+
+        //Assure init values are overwritten and that ..
+        assertEquals(serviceInfo.getServer(), "DIST500_7-F07_OC030_05_03941.local.");
+        assertEquals(serviceInfo.getPort(), 5000);
+        assertEquals(serviceInfo.getWeight(), 0);
+        assertEquals(serviceInfo.getPriority(), 0);
+        assertEquals(serviceInfo.isPersistent(), true);
+
+        // ... the ip address is set
+        assertEquals(serviceInfo.getInet4Addresses().length, 1);
+        assertArrayEquals(serviceInfo.getInet4Addresses()[0].getAddress(), new  byte[]{(byte) 192, (byte) 168,88, (byte) 236});
+    }
+
+    @SuppressWarnings("ResultOfMethodCallIgnored")
```

Comment:
I use Intellij ultimate and I want to handle all warnings the IDE gives me to let other peoble know I addressed them. Unfortentualy eclipse does not understand them.

![image](https://cloud.githubusercontent.com/assets/1171777/17244478/6005328a-5582-11e6-8b06-84e32e4c57a0.png)

![image](https://cloud.githubusercontent.com/assets/1171777/17244511/8d0e03c4-5582-11e6-98e3-fa5abcd72075.png)